src/components/StoriesGroup/StoriesGroup.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import {Modal, ModalCloseReason} from '../Modal';
+import {IndexType, StoriesLayout} from '../Stories/components/StoriesLayout/StoriesLayout';
+import {StoriesPreview} from './components';
+import {StoriesGroupItem} from './types';
+
+import {block} from '../utils/cn';
+import './StoriesGroup.scss';
+
+const b = block('stories-group');
+
+export interface StoriesGroupProps {
+    open: boolean;
+    groups: StoriesGroupItem[];
+    initialStoryIndex?: [groupIndex: number, itemIndex: number];
+    disableOutsideClick?: boolean;
+    onClose?: (
+        event: MouseEvent | KeyboardEvent | React.MouseEvent<HTMLElement, MouseEvent>,
+        reason: ModalCloseReason | 'closeButtonClick',
+    ) => void;
+    onItemSelect?: (
+        itemIndexes: [groupIndex: number, itemIndex: number],
+        selectedFromThumbnail?: boolean,
+    ) => void;
+}
+
+export const StoriesGroup = ({
+    open,
+    groups,
+    onItemSelect,
+    disableOutsideClick,
+    initialStoryIndex = [0, 0],
+    onClose,
+}: StoriesGroupProps) => {
+    const [[groupIndex, itemIndex], setCurrentStoryIndex] = React.useState(initialStoryIndex);
+
+    const handleClose = React.useCallback<NonNullable<StoriesGroupProps['onClose']>>(
+        (event, reason) => {
+            if (onClose) onClose(event, reason);
```

Comment:
Let's keep it in one style, either `if (function) function(args)`, either `function?.(args)`